sockeye/arguments.py

```diff
@@ -249,7 +249,15 @@ def add_model_parameters(params):
 
     model_params.add_argument('--weight-tying',
```

Comment:
could we rename this to 'weight-sharing' everywhere? I don't like the 'tying' expression that much anymore and its a bit fuzzy on what it means.

If this breaks backward compatibility, lets just bump the version